astropy/nddata/mixins/ndarithmetic.py

```diff
@@ -266,18 +286,21 @@ def _arithmetic(
                 operand,
                 result,
                 uncertainty_correlation,
+                axis=axis,
                 **kwds2["uncertainty"],
             )
 
         # If both are None, there is nothing to do.
-        if self.psf is not None or operand.psf is not None:
+        if self.psf is not None or (operand is not None and operand.psf is not None):
             warnings.warn(
                 f"Not setting psf attribute during {operation.__name__}.",
                 AstropyUserWarning,
             )
 
-        if handle_mask is None:
```

Comment:
Ah, perhaps it is handled here: https://github.com/astropy/astropy/pull/14175/files#diff-5057de973eaa1e5036a0bef89e618b1b03fd45a9c2952655abb656822f4ddc2aR495

Reply:
I think block needs to start with 

```python
if handle_mask is None:
    kwargs["mask"] = None
```

to ensure that setting `handle_mask` to `None` ensures no mask at the end. 

If the `handle_mask` condition is moved to the end of the conditionals, and the result is `astropy.utils.masked`, wont the result end up using the mask of `results`? 

Which, I suppose, maybe it should if the result is a `Masked` object....

Reply:
That was my intention, yes. If the result is `Masked`, it uses the mask attribute of the `Masked` object..

Reply:
I think the problem is by that setting `handle_mask` to `None` the user explicitly asking for a result with no mask. That may or may not be a sensible thing to do with one's data, but I think we need to ensure that even if the input is masked the result should have no mask (well, `.mask` should be `None`).

Reply:
BUT, maybe this is taken care of here (different than the line I tagged above because of the squash/rebase I think): https://github.com/astropy/astropy/pull/14175/files#diff-5057de973eaa1e5036a0bef89e618b1b03fd45a9c2952655abb656822f4ddc2aR493

That seems to ensure that the result's mask is `None` if `handle_mask is None`, right?